web/api/v1/api.go

```diff
@@ -1204,6 +1211,10 @@ func (api *API) metricMetadata(r *http.Request) apiFuncResult {
 					m := metadata{Type: mm.Type, Help: mm.Help, Unit: mm.Unit}
 					ms, ok := metrics[mm.Metric]
 
+					if len(metrics[mm.Metric]) >= limitPerMetric {
+						break
```

Comment:
I'm not sure if `break` is correct: see my comment about your unit test section. I may be understanding how the metadata list works, but wouldn't you have multiple, potentially different metadata types in each target? In that case, your `break` is skipping through them all.